sync3/room.go

```diff
@@ -22,7 +22,19 @@ type Room struct {
 	NumLive           int               `json:"num_live,omitempty"`
 }
 
+// RoomConnMetadata represents a room as seen by one specific connection (hence once
+// specific device).
 type RoomConnMetadata struct {
+	// We enclose copies of the data kept in the global and user caches. These snapshots
+	// represent the state we reported to the connection the last time they requested
+	// a sync. Note that we are free to tweak fields within these copies if we want, to
+	// report more appropriate data to clients that kept in the caches.
 	internal.RoomMetadata
 	caches.UserRoomData
+	// plus any per-conn data.
+
+	// LastActivityTimestamp is the largest origin_server_ts of events seen in the room
+	// that this connection is interested in. It should be less than or equal to
+	// internal.RoomMetadata.LastMessageTimestamp.
+	LastActivityTimestamp uint64
```

Comment:
You need to explain why this is in this struct and not RoomMetadata. i.e expand "that this connection is interested in" -> for bump event types.